packages/components/src/Combobox/Combobox.tsx

```diff
@@ -1,191 +1,71 @@
+import { ComboboxContextProvider } from "./ComboboxProvider";
+
+export interface ComboboxProps {
   readonly children: React.ReactNode;
 }
-interface TriggerProps {
-  /**
-   * Trigger label
-   */
-  readonly label: string;
-}
-interface ActionProps {
-  /**
-   * The function that should be performed when the action is pressed
-   */
-  readonly onClick: () => void;
-
-  /**
-   * Helper text displayed for press action
-   */
-  readonly label: string;
-}
-
-const ComboboxContext = React.createContext(
-  {} as { open: boolean; setOpen: (open: boolean) => void },
-);
 
-const ComboboxContextProvider = (props: ComboboxProps) => {
-  const [open, setOpen] = useState<boolean>(false);
-  const contentRef = useRef<HTMLDivElement>(null);
-
-  useEffect(() => {
-    function handleClickOutside(event: MouseEvent) {
-      if (
-        contentRef.current &&
-        !contentRef.current.contains(event.target as Node)
-      ) {
-        setOpen(false);
-      }
-    }
-
-    document.addEventListener("mousedown", handleClickOutside);
-    return () => {
-      document.removeEventListener("mousedown", handleClickOutside);
-    };
-  }, [contentRef]);
-
-  return (
-    <ComboboxContext.Provider value={{ open, setOpen }}>
-      <div className={styles.combobox} ref={contentRef}>
-        {props.children}
-      </div>
-    </ComboboxContext.Provider>
+export const Combobox = (props: ComboboxProps): JSX.Element => {
+  const { renderContent, renderTrigger } = useComboboxValidation(
+    props.children,
   );
-};
-
-export const Combobox = (props: ComboboxProps) => {
-  return <ComboboxContextProvider>{props.children}</ComboboxContextProvider>;
-};
-
-const TriggerButton = (props: TriggerProps) => {
-  const { open, setOpen } = React.useContext(ComboboxContext);
-
-  return <Button label={props.label} onClick={() => setOpen(!open)} />;
-};
-
-const TriggerChip = (props: TriggerProps) => {
-  const { open, setOpen } = React.useContext(ComboboxContext);
   return (
-    <div
-      className={styles.trigger}
-      onClick={() => {
-        setOpen(!open);
-      }}
-    >
-      <Text>{props.label}</Text>
-      <div className={styles.triggerIcon}>
-        <Icon name="add" size="small" />
-      </div>
-    </div>
+    <ComboboxContextProvider>
+      {renderTrigger}
+      {renderContent}
+    </ComboboxContextProvider>
   );
 };
 
-const Content = (props: ContentProps) => {
-  const { open, setOpen } = React.useContext(ComboboxContext);
-  const [searchValue, setSearchValue] = useState<string>("");
-  const [selectedOption, setSelectedOption] = useState<string>("");
-
-  const filteredOptions = props.options.filter(option =>
-    option.toLowerCase().includes(searchValue.toLowerCase()),
-  );
-
-  function handleSearch(event: React.ChangeEvent<HTMLInputElement>) {
-    setSearchValue(event.target.value);
-  }
+function useComboboxValidation(children: ReactNode): {
+  renderTrigger: ReactNode;
+  renderContent: ReactNode;
```

Comment:
thoughts on calling them thangs as `triggerElement` and `contentElement` it isn't really trying to render but more of finding and returning an element.

Reply:
I'm not too opinionated there. I've seen variable naming that prefixes with `render` for stuff like this, but I get what you're saying, technically it's already been rendered out where you invoked this component with `Combobox.Content` and at this point, we're dealing with the element

with that in mind, I'm fine with changing it to `blahElement`